### ISSUE TITLE
Use a consistent example and type naming throughout the spec

### ISSUE THREAD

**Issue by privatwolke** [json-api/json-api#941]
I am implementing JSON API server side and try to validate my implementation by transcribing the spec to unit tests as closely as possible. Doing this has shown me that the spec uses many different example resources (articles, photos, comments, person, people, ...).

To make things easier to understand I would propose to settle on one example data model which is expanded to explain the relevant sections of the spec as needed.

**Comment by jelhan** [comment]
I don't think we can settle on one data model. Using only one resource type would make if difficult to demonstrate features like compound documents, relationship links, inclusion of related resources and spare fieldsets.

The specification uses the following resource types in fetching section: `articles`, `people`, `comments` and `tags`.

`articles` is used as the main resource type. `people` is used to demonstrate to-one and to-many relationships. `comments` and `tags` are both used to demonstrate to-many relationships. `tags` is used for relationship links.

In most cases `comments` is used in examples for to-many relationships. `tags` is used only in three cases:

1. [8.2 Fetching relationships](https://jsonapi.org/format/1.1/#fetching-relationships)
2. [9.2.2 Updating a Resource’s Relationships](https://jsonapi.org/format/1.1/#crud-updating-resource-relationships)
3. [9.3  Updating Relationships](Updating Relationships)

For that topics, it seems to be relevant that a reader expects an relationship of an article and tags to change. The example would feel made up if using `comments` instead.

Looking at the use cases for `comments`, I haven't noticed any reason, why it couldn't be replaced with `tags`. Doing so would reduce the number of different resource types a little bit.

In the part about [creating resources](https://jsonapi.org/format/1.1/#crud-creating), a `photos` resource type is used in the example. This is surprising. Especially as the section about [updating resources](https://jsonapi.org/format/1.1/#crud-updating) used the types introduced for fetching already. I think the `photos` type can be safely replaced by `articles` type.

I guess it's time for two pull requests. 😅
